src/app/tasks/components/TaskLoggerActions/TaskLoggerActions.tsx

```diff
@@ -19,7 +22,28 @@ type TaskLoggerActionsProps = {
   isUploadTask: boolean;
 };
 
+const pauseUpload = (id: string) => {
+  localStorageService.setUploadState(id, TaskStatus.Paused);
+  tasksService.updateTask({
+    taskId: id,
+    merge: { status: TaskStatus.Paused },
+  });
+};
+
+const resumeUpload = (id: string) => {
+  localStorageService.setUploadState(id, TaskStatus.InProcess);
```

Comment:
I recommend using the local database rather than the local storage, also, if we can hide which is the implementation under a cache service, the better. 

If we do not do that, what is going to happen if the local storage suddenly is not supported, is blocked or we want to switch to another persistence layer? We would need to change each local storage usage in the code, instead of having it centralized. I think the Repository pattern is the right fit here